packages/backend-defaults/src/entrypoints/urlReader/lib/AzureBlobStorageUrlReader.ts

```diff
@@ -0,0 +1,244 @@
+    });
+  };
+
+  // private readonly blobServiceClient: BlobServiceClient;
+
+  constructor(
+    private readonly credsManager: AzureCredentialsManager,
+    private readonly integration: AzureBlobStorageIntergation,
+    private readonly deps: {
+      treeResponseFactory: ReadTreeResponseFactory;
+    },
+  ) {}
+
+  private async createContainerClient(
+    containerName: string,
+  ): Promise<ContainerClient> {
+    const accountName = this.integration.config.accountName; // Use the account name from the integration config
+    const accountKey = this.integration.config.accountKey; // Get the account key if it exists
+
+    if (accountKey && accountName) {
+      const creds = new StorageSharedKeyCredential(accountName, accountKey);
+      const blobServiceClient = new BlobServiceClient(
+        `https://${accountName}.${this.integration.config.host}`,
+        creds,
+      );
+      return blobServiceClient.getContainerClient(containerName);
+    }
+    // Use the credentials manager to get the correct credentials
+    const credential = await this.credsManager.getCredentials(
+      accountName as string,
+    );
+
+    let blobServiceClientUrl: string;
+
+    if (this.integration.config.endpoint) {
+      if (this.integration.config.sasToken) {
+        blobServiceClientUrl = `${this.integration.config.endpoint}?${this.integration.config.sasToken}`;
+      } else {
+        blobServiceClientUrl = `${this.integration.config.endpoint}`;
+      }
+    } else {
+      blobServiceClientUrl = `https://${this.integration.config.accountName}.${this.integration.config.host}`;
+    }
+
+    const blobServiceClient = new BlobServiceClient(
+      blobServiceClientUrl,
+      credential,
+    );
+    return blobServiceClient.getContainerClient(containerName);
+  }
+
+  async read(url: string): Promise<Buffer> {
+    const response = await this.readUrl(url);
+    return response.buffer();
+  }
+
+  async readUrl(
+    url: string,
+    options?: UrlReaderServiceReadUrlOptions,
+  ): Promise<UrlReaderServiceReadUrlResponse> {
+    const { etag, lastModifiedAfter } = options ?? {};
+
+    try {
+      const { path, container } = parseUrl(url);
+
+      const containerClient = await this.createContainerClient(container);
+      const blobClient = containerClient.getBlobClient(path);
+
+      const abortController = new AbortController();
+
+      const getBlobOptions = {
+        conditions: {
+          ...(etag && { ifNoneMatch: etag }),
+          ...(lastModifiedAfter && { ifModifiedSince: lastModifiedAfter }),
+        },
+      };
+      options?.signal?.addEventListener('abort', () => abortController.abort());
+
+      const downloadBlockBlobResponse = await blobClient.download(
+        0,
+        undefined,
+        getBlobOptions,
+      );
+
+      const data = await this.retrieveAzureBlobData(
+        downloadBlockBlobResponse.readableStreamBody as Readable,
+      );
+
+      return ReadUrlResponseFactory.fromReadable(data, {
```

Comment:
do we need this `retrieveAzureBlobData`? The issue is that it's pushing all the chunks in memory which could cause some memory issues.

```suggestion
      return ReadUrlResponseFactory.fromReadable(downloadBlockBlobResponse.readableStreamBody, {
```